src/lib/token-purchase.ts

```diff
@@ -0,0 +1,63 @@
+import type {Name} from '@greymass/eosio'
+
+import {addToast} from '~/stores/toast'
+
+const creationServiceUrl = import.meta.env.SNOWPACK_PUBLIC_WHALESPLAINER_URL
+const unicoveUrl = import.meta.env.SNOWPACK_PUBLIC_UNICOVE_URL
+
+const tokenOrderUrl = `${creationServiceUrl}/api/tokens/order`
+
+export const openPopupForAccount = async (accountName: Name | undefined): Promise<void> => {
+    let whalesplainerTokenOrderResponse
+    let whalesplainerTokenOrder
+
+    try {
+        whalesplainerTokenOrderResponse = await fetch(tokenOrderUrl, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                wallet_address: accountName?.toString(),
+                account_reference: accountName?.toString(),
+                fiat_code: 'USD',
+                coin_code: 'EOS',
+                return_url_on_success: unicoveUrl,
+            }),
+        })
+
+        whalesplainerTokenOrder = await whalesplainerTokenOrderResponse.json()
+
+        console.log({whalesplainerTokenOrder})
+    } catch (error) {
```

Comment:
If this throws the toast below will fire as well, possibly replacing this error message with a generic one about popups.

Refactor this so that error handling only needs to happen once, preferably at the callsite and not side-effecting with a toast